### PR TITLE
Additional unit test

### PR DESCRIPTION
**Pull request by domanttas**
Added unit tests for IsIdentified

### CODE REVIEW

**Comment by tomasmuzas** [comment]
I didn't notice that while reviewing your previous PR. I just noticed that your method accepts all strings which may or may not be names. It would probably be better to identify success by response status and then send additional info as error code.
An example from our code:
```
Status: 404 (just an example, it could be 400 or other fitting status)
{
   error: "face.not_detected",
   message: "no face was detected due to poor lighting"
}
```
This way you dont rely upon strings to know that something failed, but if you dont get 2XX or 3XX status code, you know something went wrong, and also, you have additional info what happened provided as error code

**Comment by tomasmuzas** [comment]
I have seen you are doing something similar, I will probably take a look at your whole code to see a full context

**Comment by domanttas** [comment]
So I should create some structure for errors and related messages and based on return status code use them to decide what actions are required? I was relying on strings because as I created my Web API I structured it so that it would returns these strings when call is successful. I'm really not sure how to handle responses and errors in a good way tho, I was trying to modify it as you mentioned in previous PR's, so I appreciate the advice

**Comment by domanttas** [comment]
If you would like to take a look at my API, I have already added you to another repo, I think. But I understand that it takes a lot of time from you, so I don't want to ask you for too much, you are already putting in a lot of effort in this :D

**Comment by domanttas** [comment]
You weren't added, apparently. I was pretty sure that I've sent invite for you few weeks ago, my bad

**Comment by tomasmuzas** [comment]
You are on the right track! There are just two things:
* it is better to avoid special symbols like spaces and ! to make it more sentence-like
* there may be many unrelated fails, for example `500` (it's unavoidable :D), `503 not authirized`, `404` etc.
So it's better to have some logic relying upon statuses, unless you are handling all errors as fatal and display something generic as "something went wrong"
Also one status code can have more reasons then

**Comment by domanttas** [comment]
500 scares me a lot :D Thank you, I will try to modify handling when will find the time for it as you guided me. Just few quick questions, if it doesn't bother you: is it bad practise to handle all errors as fatal and should I throw exception when I catch bad status code or try to re - call API/display some message for user?  (I don't really feel comfortable when throwing exceptions, especially in frontend)

**Comment by tomasmuzas** [comment]
Let me tell you a story... :DD
Talking about separating fatal and non fatal errors: 500 is definitely a fatal one, and unplanned. Then there are other error codes. There comes that structure into play. If you can parse it as an error and act according to the error code, it is an expected behaviour and you should know how to handle that. If you cannot parse it (lets say you get `404` by calling wrong endpoint by mistake), then this is fatal too.

Now about the exceptions. Thats much broader topic. Ideally, you should not throw exceptions unless there is an unhandled case:
```
catch (ConcreteException1 ex)
catch (SomeOtherException ex)
catch (Exception ex) // that was unexpected, so you can and probably should rethrow it
```
Ideally, you should have a global exception handler (which is not that easy to implement) which logs it and at the end just wraps it into `500` with generic error code and returns it to the frontend. It also means that you shouldnt try/catch all your code and future proof for any possible exception that may or may not happen. Exception can also be unrelated to the request/response, so even try/catch wont save you.

In short, backend should have predefined expected cases where code can fail and then have a contract with frontend whether or not it cares what specifically happened (a good example would be, lets say, that a monetary transaction failed, so any error is equally fatal) and then you should return error accordingly. Also, frontend should be able to handle generic error `500`

**Comment by tomasmuzas** [comment]
If you want, we can meet some day, and take a deeper look at your use cases

**Comment by domanttas** [comment]
Thanks for an explanation. I need to wrap my mind around this :D 

I don't really want to bother you any more, you are investing a lot of time into me already, however I know that meeting with you would be extremely beneficial for me and would help me a lot.

Then again, I really don't want to use up your time for something like random second year student's home assignment :D

**Comment by tomasmuzas** [comment]
Well I'm offering it myself and willingly

**Comment by domanttas** [comment]
Okay, it would be really cool then. Let me know when you will be available
